### PR TITLE
CLI parameters for the -g and -w options did not output to .pka out file

### PR DESCRIPTION
**Pull request by danny305**
I was trying to get more granular free energy and protein charge output in the .pka file. However, passing in different parameters to the -g and -w CLI parameters did not modify the out .pka file. 

I made changes in the source code to address this and solved some floating point rounding issues along the way (used the decimal library). 

All tests passed.

### CODE REVIEW

**Review by sobolevnrm — APPROVED** [review]
This looks good to me but I'd like @speleo3 or @orbeckst to also review before we merge.  Thanks!

File: propka/output.py [hunk 1]
```diff
@@ -285,13 +286,16 @@ def get_folding_profile_section(
     profile, [ph_opt, dg_opt], [dg_min, dg_max], [ph_min, ph_max] = (
         protein.get_folding_profile(
             conformation=conformation, reference=reference,
-            grid=[0., 14., 0.1]))
+            grid=protein.options.grid))
+
     if profile is None:
         str_ += "Could not determine folding profile\n"
     else:
+        delta = Decimal(str(round(window[2],2)))
```
Comment (speleo3) [inline propka/output.py:294]: `str` is not needed, you can do this:
```python
delta = round(Decimal(window[2]), 2)
```

Comment (orbeckst) [reply propka/output.py:294]: Is there a reason why `Decimal` as opposed to just using `float`?

Comment (danny305) [reply propka/output.py:294]: Yeah so if I do not cast to a string and then to a Decimal then I get precision/rounding errors. I tried floats and then I tried just Decimal without the str cast and the bug persisted. This method solved the issues on a consistent basis.

Comment (orbeckst) [reply propka/output.py:294]: Can you elaborate: What did you expect to happen and what happened instead (without Decimal)?

Comment (danny305) [reply propka/output.py:294]: Sure.
Here is the command I will provide the example with: 
`python -m propka -g 7 9 0.1 -w 7 9 0.1 6ij6.pdb`
And here is a snippet from the output.pka file this PR will give you (which is what should be expected):

Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)
  7.00     21.83
  7.10     21.69
  7.20     21.54
  7.30     21.40
  7.40     21.25
  7.50     21.09
  7.60     20.94
  7.70     20.77
  7.80     20.61
  7.90     20.44
  8.00     20.26
  8.10     20.08
  8.20     19.90
  8.30     19.71
  8.40     19.52
  8.50     19.32
  8.60     19.12
  8.70     18.91
  8.80     18.70
  8.90     18.48
  9.00     18.26
The pH of optimum stability is  9.0 for which the free energy is   18.3 kcal/mol at 298K
Could not determine pH values where the free energy is within 80 % of minimum
Could not determine the pH-range where the free energy is negative


Here is the output if I remove the string casting within the Decimal constructor on line 295 and 297:
```
else:
        delta = Decimal(round(window[2],2))    # Line 295
        for (ph, dg) in profile:
            ph = Decimal(round(ph, 3))                # Line 297
            if ph >= window[0] and ph <= window[1]:
                if ph % delta < 0.05 or ph % delta > 0.95:
                    str_ += "{0:>6.2f}{1:>10.2f}\n".format(ph, dg)
        str_ += "\n"

```
Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)
  8.30     19.71
  8.80     18.70
The pH of optimum stability is  9.0 for which the free energy is   18.3 kcal/mol at 298K
Could not determine pH values where the free energy is within 80 % of minimum
Could not determine the pH-range where the free energy is negative


Here is the output if we leave `delta` and `ph` as floats:
```
else:
        delta = window[2]       # Line 295
        for (ph, dg) in profile:
            # ph = Decimal(round(ph, 3))       # Line 297
            if ph >= window[0] and ph <= window[1]:
                if ph % delta < 0.05 or ph % delta > 0.95:
                    str_ += "{0:>6.2f}{1:>10.2f}\n".format(ph, dg)
        str_ += "\n"
```
Free energy of   folding (kcal/mol) as a function of pH (using neutral reference)

The pH of optimum stability is  9.0 for which the free energy is   18.3 kcal/mol at 298K
Could not determine pH values where the free energy is within 80 % of minimum
Could not determine the pH-range where the free energy is negative


As you can clearly see. I only get the expected output for a specified window and grid  when I cast the floats to a string and then turn them into a Decimal object.

Comment (speleo3) [reply propka/output.py:294]: Thanks @danny305 for the example. As I wrote in my first comment, you don't need `str` if you round after converting to Decimal.
```python
>>> Decimal(str(round(math.pi, 2))) == round(Decimal(math.pi), 2)
True
```
The following two give the same result:
1. round -> str -> Decimal
2. Decimal -> round

Comment (danny305) [reply propka/output.py:294]: Interesting. I can update the PR branch.

**Review by speleo3 — APPROVED** [review]

**Review by sobolevnrm — APPROVED** [review]

**Comment by danny305** [comment]
what is needed for this to be merged???

**Comment by intendo** [comment]
I think @orbeckst still needs to review this before we can merge this PR.

**Review by orbeckst — APPROVED** [review]
I am not a big fan of rounding any input but looking at what the code does (in particularly, looking at hardcoded 0.05 spacing I am ok with the changes.

Sorry for the delayed response. In the future please feel free to ping me again.

**Pull request merged**